paddle/memory/malloc.h

```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+
+#include <paddle/majel/place.h>
+
+namespace paddle {
+namespace memory {
+
+void init();
+void shutdown();
+
+void* malloc(majel::Place place, size_t size);
```

Comment:
How about we add the following two functions here:

```cpp
malloc_pinned_and_mapped(size_t host_memory_size);
malloc_pinned(size_t host_memory_size);
```

so we can write 

1. GpuAllocator, which corresponds to Majels' `cpu/SystemAllocator`, 
1. GPUAllocator, which corresponds to Majel's `gpu/SystemAllocaotr`, and
1. BuddyAllocator, which corresponds to Majels' `detail/BuddyAllocator`, which fail backs to either GPUAllocator or CPUAllocaotr.